apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/controller/ConfigsExportController.java

```diff
@@ -1,91 +1,55 @@
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
-import org.springframework.web.multipart.MultipartFile;
-
-import javax.servlet.http.HttpServletResponse;
-import java.io.IOException;
-import java.io.InputStream;
-import java.util.List;
-import java.util.Objects;
-import java.util.stream.Collectors;
 
 /**
  * jian.tan
  */
 @RestController
 public class ConfigsExportController {
 
-  private final ItemService configService;
+  private final ConfigsExportService configsExportService;
 
   private final NamespaceService namespaceService;
 
   public ConfigsExportController(
-      final ItemService configService,
-      final @Lazy NamespaceService namespaceService) {
-    this.configService = configService;
+      final ConfigsExportService configsExportService,
+      final @Lazy NamespaceService namespaceService
+  ) {
+    this.configsExportService = configsExportService;
     this.namespaceService = namespaceService;
   }
 
-  @PreAuthorize(value = "@permissionValidator.hasModifyNamespacePermission(#appId, #namespaceName, #env)")
-  @PostMapping("/apps/{appId}/envs/{env}/clusters/{clusterName}/namespaces/{namespaceName}/items/import")
-  public void importConfigFile(@PathVariable String appId, @PathVariable String env,
-      @PathVariable String clusterName, @PathVariable String namespaceName,
-      @RequestParam("file") MultipartFile file) {
-    if (file.isEmpty()) {
-      throw new BadRequestException("The file is empty.");
-    }
-
-    NamespaceDTO namespaceDTO = namespaceService
-        .loadNamespaceBaseInfo(appId, Env.fromString(env), clusterName, namespaceName);
-
-    if (Objects.isNull(namespaceDTO)) {
-      throw new BadRequestException(String.format("Namespace: %s not exist.", namespaceName));
-    }
-
-    NamespaceTextModel model = new NamespaceTextModel();
-    List<String> fileNameSplit = Splitter.on(".").splitToList(file.getOriginalFilename());
-    if (fileNameSplit.size() <= 1) {
-      throw new BadRequestException("The file format is invalid.");
-    }
-
-    String format = fileNameSplit.get(fileNameSplit.size() - 1);
-    model.setFormat(format);
-    model.setAppId(appId);
-    model.setEnv(env);
-    model.setClusterName(clusterName);
-    model.setNamespaceName(namespaceName);
-    model.setNamespaceId(namespaceDTO.getId());
-    String configText;
-    try(InputStream in = file.getInputStream()){
-      configText = ConfigToFileUtils.fileToString(in);
-    }catch (IOException e) {
-      throw new ServiceException("Read config file errors:{}", e);
-    }
-    model.setConfigText(configText);
-
-    configService.updateConfigItemByText(model);
-  }
-
+  /**
+   * export one config as file.
+   * keep compatibility.
+   * file name examples:
+   * <pre>
+   *   application.properties
+   *   application.yml
+   *   application.json
+   * </pre>
+   */
+  @PreAuthorize(value = "@permissionValidator.isAppAdmin(#appId)")
```

Comment:
this permission should check `PermissionValidator#shouldHideConfigToCurrentUser`